packages/appeals-service-api/src/services/back-office-v2/validate.js

```diff
@@ -0,0 +1,81 @@
+const Ajv = require('ajv').default;
+const { loadAllSchemas } = require('pins-data-model');
+
+/**
+ * @typedef {import('pins-data-model').LoadedSchemas} LoadedSchemas
+ */
+
+/**
+ * @template Payload
+ * @typedef {{(schema: string, payload: unknown): payload is Payload}} ValidateFromGivenSchema
+ */
+
+/**
+ * @template Payload
+ * @typedef {{(payload: unknown): payload is Payload}} Validate
+ */
+
+/**
+ * @template Payload
+ * @type {(commandsAndEvents: LoadedSchemas) => ValidateFromGivenSchema<Payload>}
+ */
+
+// This class is used to obfuscate the async nature of loadAllSchemas
+// Clear it out once the schemas can be loaded synchronously
```

Comment:
didn't you achieve this?

Reply:
I did but I wrote this thing first 😅

Reply:
turns out loadAllSchemasSync isn't that much more useful here because I needed to dynamically import the data model module so it's a promise anyway